itests/supply_test.go

```diff
@@ -0,0 +1,204 @@
+		require.NoError(t, err)
+		fullNode0.WaitMsg(ctx, c00)
+		c01, err := blockMiner0.FullNode.MarketAddBalance(ctx, worker0, blockMiner0.ActorAddr, lockedProviderBalance)
+		require.NoError(t, err)
+		fullNode0.WaitMsg(ctx, c01)
+
+		psd0, err := fullNode0.MpoolPushMessage(ctx,
+			makePSDMessage(
+				t,
+				ctx,
+				blockMiner0.ActorAddr,
+				worker0,
+				wallet0,
+				lockedProviderBalance,
+				lockedClientBalance,
+				fullNode0.WalletSign,
+			),
+			nil,
+		)
+		require.NoError(t, err)
+		fullNode0.WaitMsg(ctx, psd0.Cid())
+		head, err := fullNode0.ChainHead(ctx)
+		require.NoError(t, err)
+		height0 = head.Height()
+	}
+
+	// Lock collateral in market on nv23 network
+	fullNode1, blockMiner1, ens1 := kit.EnsembleMinimal(t,
+		kit.GenesisNetworkVersion(network.Version23),
+		kit.MockProofs(),
+	)
+	{
+		worker1 := blockMiner1.OwnerKey.Address
+		ens1.InterconnectAll().BeginMining(50 * time.Millisecond)
+
+		// Lock collateral in market actor
+		wallet1, err := fullNode1.WalletDefaultAddress(ctx)
+		require.NoError(t, err)
+		c10, err := fullNode1.MarketAddBalance(ctx, wallet1, wallet1, lockedClientBalance)
+		require.NoError(t, err)
+		fullNode1.WaitMsg(ctx, c10)
+		c11, err := blockMiner1.FullNode.MarketAddBalance(ctx, worker1, blockMiner1.ActorAddr, lockedProviderBalance)
+		require.NoError(t, err)
+		fullNode1.WaitMsg(ctx, c11)
+
+		psd1, err := fullNode1.MpoolPushMessage(ctx,
+			makePSDMessage(
+				t,
+				ctx,
+				blockMiner1.ActorAddr,
+				worker1,
+				wallet1,
+				lockedProviderBalance,
+				lockedClientBalance,
+				fullNode1.WalletSign,
+			),
+			nil,
+		)
+		require.NoError(t, err)
+		fullNode1.WaitMsg(ctx, psd1.Cid())
+		head, err := fullNode1.ChainHead(ctx)
+		require.NoError(t, err)
+		height1 = head.Height()
+	}
+
+	// Measure each circulating supply at the latest height where market balance was locked
+	// This allows us to normalize against fluctuations in circulating supply based on the underlying
+	// dynamics irrelevant to this change
+
+	max := height0
+	if height0 < height1 {
+		max = height1
+	}
+	max = max + 1 // Measure supply at height after the deal locking funds was published
+
+	// Let both chains catch up
+	fullNode0.WaitTillChain(ctx, func(ts *types.TipSet) bool {
+		return ts.Height() >= max
+	})
+	fullNode1.WaitTillChain(ctx, func(ts *types.TipSet) bool {
+		return ts.Height() >= max
+	})
+
+	ts0, err := fullNode0.ChainGetTipSetByHeight(ctx, max, types.EmptyTSK)
+	require.NoError(t, err)
+	ts1, err := fullNode1.ChainGetTipSetByHeight(ctx, max, types.EmptyTSK)
+	require.NoError(t, err)
+
+	nv22Supply, err := fullNode0.StateVMCirculatingSupplyInternal(ctx, ts0.Key())
+	require.NoError(t, err, "Failed to fetch nv22 circulating supply")
+	nv23Supply, err := fullNode1.StateVMCirculatingSupplyInternal(ctx, ts1.Key())
+	require.NoError(t, err, "Failed to fetch nv23 circulating supply")
+
+	// Unfortunately there's still some non-determinism in supply dynamics so check for equality within a tolerance
```

Comment:
This is a huge pain and I'm not sure where the non-determinism is coming from -- we are measuring two identically set up test networks with a single miner at the exact same epoch.  If I had to guess I'd say sensitivity of IP function to variations in message landing time.

Though I would love to understand why it is happening and eliminate non determinism from the itest framework, for the purposes of testing FIP 0065 isolating out the non-determinism is probably a lot of work for something that is ultimately not important.

Reply:
that's kind of weird, is the difference unstable across different test runs? this wouldn't be due to the f099 changes would it?

Reply:
Any ensemble run has the difference.  So between test runs and between the two ensembles in every test run.

Tbh it's not particularly surprising from the reputation of economics in new networks being difficult to work with.  And the sensitivity to message timing of filtering on power and circ supply[ within actors](https://github.com/filecoin-project/ref-fvm/blob/d10baf82857df234b37404507df8bbaa581990ac/shared/src/smooth/alpha_beta_filter.rs) is a hypothesis I'm reasonably ok with.

Reply:
The difference in locked balance is almost certainly coming from differences in total InitialPledge between runs as Burnt is kept in a different category.